tools/Dockerfiles/fedora_sandbox

```diff
@@ -0,0 +1,41 @@
+FROM fedora:rawhide
+
+# Install generic dependencies to build jss
+RUN true \
+        && dnf update -y --refresh \
+        && dnf install -y dnf-plugins-core gcc make rpm-build cmake \
+                          glassfish-jaxb-api nss-tools apache-commons-codec \
+                          apache-commons-lang gcc-c++ jpackage-utils slf4j \
+                          zlib-devel perl slf4j-jdk14 junit ninja-build \
+                          gyp gtest mercurial \
+        && dnf build-dep nss nspr jss -y \
+        && mkdir -p /home/sandbox \
+        && dnf clean -y all \
+        && rm -rf /usr/share/doc /usr/share/doc-base \
+                  /usr/share/man /usr/share/locale /usr/share/zoneinfo \
+        && true
+
+# Link in the current version of jss from the git repository
+WORKDIR /home/sandbox
+COPY . /home/sandbox/jss
+
+# Download and build NSPR and NSS
+RUN true \
+        && cd /home/sandbox \
+        && hg clone https://hg.mozilla.org/projects/nspr \
+        && hg clone https://hg.mozilla.org/projects/nss \
+        && cd nss \
+        && ./build.sh --enable-fips --enable-libpkix \
```

Comment:
Question: After building nss, do you need to install or let jss know which nss to use?

Also, in the description, it says that both `nspr` and `nss` are built. But, I see only nss to be built. Doesn't `nspr` require to be built?

Reply:
NSS's build also builds NSPR from the same sandbox by default. You have to specify additional arguments to get it to use the system NSPR. That has actually failed for me sometimes.

By specifying `export SANDBOX=1`, JSS will pull NSS and NSPR from the sandbox, so you don't have to install it in order to use it.

Reply:
To clarify: a sandbox is a directory structure like:

```
some_dir/
    nspr/
    nss/
    jss/
```

After building nss, you'll also get a `some_dir/dist/` folder. When building JSS with `SANDBOX=1` exported, it'll look in the parent directory for a dist folder and use NSS/NSPR from the built distribution.